analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/ValidatorTestCheck.cs

```diff
@@ -99,6 +99,9 @@ protected override ProgramState PostProcessSimple(SymbolicContext context)
             }
             else if (context.Operation.Instance is IInvocationOperation invocation && invocation.TargetMethod.Name == "Tag")
             {
+                (invocation.TargetMethod.IsStatic
+                    || invocation.TargetMethod.IsExtensionMethod
+                    || invocation.Arguments.Length == 1).Should().BeTrue("Tag method should be static to not infer with object states.");
```

Comment:
We do similar checks a few lines down too `tagName.HasValue.Should().BeTrue("tag should have literal name");` I used this to find any violations and we can keep it to never introduce subtle hard-to-diagnose bugs.

Reply:
Why is there `IsExtensionMethod`? Do we have at least one? Otherwise, I'd remove it.
Why is there `|| invocation.Arguments.Length == 1`? It's inconsistent with what the rest asserts. I'd remove it

Reply:
We have one test case that does `someSample.Tag()`. I removed the `IsExtensionMethod` check